askup/utils/views.py

```diff
@@ -284,3 +289,36 @@ def apply_filter_to_queryset(request, filter, queryset):
         return queryset.exclude(user_id=request.user.id)
 
     return queryset
+
+
+def compose_qset_user_questions_response(queryset):
+    """
+    Compose qset user questions list from the queryset.
+    """
+    questions = []
+
+    for question in queryset:
+        questions.append([question.id, question.text, question.vote_value])
+
+    return questions
+
+
+def get_user_subjects(user_id):
+    """
+    Return a list of prepared user subjects data.
+    """
+    queryset = get_user_subjects_queryset(user_id)
+    subjects = compose_user_qsets(queryset)
+    return subjects
+
+
+def compose_user_qsets(queryset):
+    """
+    Compose user qsets list from the questions queryset.
+    """
+    qsets = []
+
+    for question in queryset:
```

Comment:
This function could be avoided with the using `values_list()` in the function `get_user_subjects`.

Reply:
done